agent/connect/ca/provider_consul.go

```diff
@@ -31,94 +34,109 @@ type ConsulProviderStateDelegate interface {
 
 // NewConsulProvider returns a new instance of the Consul CA provider,
 // bootstrapping its state in the state store necessary
-func NewConsulProvider(rawConfig map[string]interface{}, delegate ConsulProviderStateDelegate) (*ConsulProvider, error) {
-	conf, err := ParseConsulCAConfig(rawConfig)
-	if err != nil {
-		return nil, err
-	}
-	provider := &ConsulProvider{
-		config:   conf,
+func NewConsulProvider(delegate ConsulProviderStateDelegate) *ConsulProvider {
+	return &ConsulProvider{
 		delegate: delegate,
-		id:       fmt.Sprintf("%s,%s", conf.PrivateKey, conf.RootCert),
 	}
+}
+
+func (c *ConsulProvider) Configure(clusterId string, isRoot bool, rawConfig map[string]interface{}) error {
+	// Parse the raw config and update our ID.
+	config, err := ParseConsulCAConfig(rawConfig)
+	if err != nil {
+		return err
+	}
+	c.config = config
+	c.id = fmt.Sprintf("%s,%s", config.PrivateKey, config.RootCert)
```

Comment:
Hashing seems like a good idea - I wonder what the best way to do this is without breaking upgrades (as the new hash will point to a non-existent entry and trigger a cert mismatch with the current active root). Maybe we leave the current method in for a major version and add some logic to use/write to the new hash ID alongside it when it detects an old-style ID?

Reply:
Can we just check if the old key exists if the hashed key doesn't?

Honestly I missed that this was not a change and we are already doing this in state so good question but I think it's simple to "live migrate" in this configure call rather than take a breaking update.